### PR TITLE
[BD-29] [TNL-7060]: Add titles to course search indexing for all xblocks

### PR DESCRIPTION
**Pull request by ahmedbilal**
Modify XBlock root class's index_dictionary method to return common fields e.g `title` and `content_type` by default. 

**JIRA tickets**:
* https://openedx.atlassian.net/browse/TNL-7060

**Dependencies**: None

**Merge deadline**: "None"

**Testing instructions**:

None yet.

**Author notes and concerns**:

1. I am not sure what kind of unit test should I write to test this feature as all the real XBlocks lives in other repos. So, if you have some idea, please discuss.
 
**Reviewers**
- [ ] @xitij2000
- [ ] @davestgermain

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ahmedbilal! I've created [OSPR-4778](https://openedx.atlassian.net/browse/OSPR-4778) to keep track of it in JIRA, where we prioritize reviews.  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@ahmedbilal Thank you for your contribution. Please let me know once it is ready for our review.

File: xblock/mixins.py [hunk 1]
```diff
@@ -556,9 +556,22 @@ def index_dictionary(self):
...
+        content_type = getattr(
+            getattr(self.__class__, "display_name", None), "default", "Component"
+        )
+
+        return {
+            "content_type": content_type,  # e.g ORA, Video, HTML etc.
+            "content": {
+                "display_name": display_name
+            }
+        }
```
Comment (xitij2000) [inline xblock/mixins.py:574]: I don't think having a default here makes sense. Just don't add these keys to the index if the value is `None`. I don't currently know of a reason why they would be, but still, a default value of "Component", doesn't make sense to me and it also doesn't make sense to index a "None" value.

Comment (ahmedbilal) [reply xblock/mixins.py]: @xitij2000 Fixed

File: xblock/VERSION.txt [hunk 2]
```diff
@@ -1 +1 @@
-1.3.1
+1.4
```
Comment (xitij2000) [inline xblock/VERSION.txt:1]: nit: this should be `1.4.0`

Comment (ahmedbilal) [reply xblock/VERSION.txt]: @xitij2000 Done. Should I ask natabene for review?

**Comment by xitij2000** [comment]
@ahmedbilal Please prefix [BD-29] [TNL-7060] to the PR title so it is tagged properly.

**Comment by ahmedbilal** [comment]
> @ahmedbilal Please prefix [BD-29] [TNL-7060](https://openedx.atlassian.net/browse/TNL-7060) to the PR title so it is tagged properly.

Done

**Comment by ahmedbilal** [comment]
@natabene This is ready for review

**Comment by natabene** [comment]
@ahmedbilal Since it is a BD ticket, it will now be processed accordingly.

**Comment by ahmedbilal** [comment]
@davestgermain Can you be able to review this PR? It is related to all those indexing PRs.

**Comment by xitij2000** [comment]
@davestgermain  Could you tag a release for this as well?

**Comment by davestgermain** [comment]
@xitij2000 done

**Pull request merged**